storage/filesystem/object.go

```diff
@@ -36,9 +37,66 @@ func (s *ObjectStorage) Set(core.Object) (core.Hash, error) {
 	return core.ZeroHash, fmt.Errorf("not implemented yet")
 }
 
+func (s *ObjectStorage) Get(h core.Hash) (core.Object, error) {
+	var obj core.Object
+	var err error
+
+	obj, err = s.getFromObject(h)
+	if err == nil {
```

Comment:
It is ok to look for the object in the packfile if the object has not been found in the objects dir, but what if getFromObject fails for other reasons? (for example if we don't have the permissions or because the connection with the (remote) file system has fail).

We must do proper error checking of the return values of getFromObject, defining specific error for when the object has not been found.